### DISCUSSION TITLE
Is there an open-source alternative to the Dev Containers extension?

### DISCUSSION THREAD

**Discussion by schuelermine** [VSCodium/vscodium#2048] [Q&A]
Specifically, this one: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers (ms-vscode-remote.remote-containers)

There is an open-source CLI for spinning up these containers but not for connecting VSCodium to them.

**Comment by schuelermine** [comment]
Or, I guess, more generally, an extension to connect to containers (without SSH). Making the containers seems to be possible with the CLI… unless the extension would need to inject something into the container spec on build.

**Reply by craigphicks** [reply]
Just out of curiosity, why do you specify "without SSH"?  If you use the extension `Open Remote - SSH` [by jeanp413], it will, via ssh, connect to the specified remote and try to install vscodium remote server software on the target.  That is very close to the behavior of Remote Dev Containers , I think.
